alpha_automl/automl_api.py

```diff
@@ -393,3 +393,59 @@ def _column_parser(self, X):
     def fit(self, X, y=None):
         X, y = self._column_parser(X)
         super().fit(X, y)
+
+
+class AutoMLSemiSupervisedClassifier(BaseAutoML):
+
+    def __init__(self, output_folder, time_bound=15, metric='f1_score', split_strategy='holdout',
+                 time_bound_run=5, score_sorting='auto', metric_kwargs={'average': 'micro'}, split_strategy_kwargs=None,
+                 start_mode='auto', verbose=False):
+        """
+        Create/instantiate an AutoMLSemiSupervisedClassifier object.
```

Comment:
It is done by Roque on this commit 9f998633da9d7a91864127a9cc1dfafa468d062c.
@roquelopez Do we want to move this script?

Reply:
The idea of having a `demo_utils` folder was to group all the code needed to run Alpha-AutoML in the NYU HSRN server. For instance, in values.yaml file, there are some [configs](https://github.com/VIDA-NYU/alpha-automl/blob/devel/kubernetes/values.yaml#L32-L48) specifically for NYU HSRN server. This will not work in other Kubernetes servers. That is why I prefer to have a `demo` to a `kubernetes` folder. However, if we can add documentation to set up these configs, I think it's ok to have the `kubenetes` folder. 
Also, why do we need [jupyterhub, jupyterlab, and jupyterlab-server ](https://github.com/VIDA-NYU/alpha-automl/blob/devel/Dockerfile#L5) in the Dockerfile? If it is a 'generic' Dockerfile only the `notebook` dependency is enough. By generic I mean that users can run it in any kind of environment, not exclusively in Kubernetes. I think that jupyterhub, jupyterlab, and jupyterlab-server dependencies were added there because they are needed by the Jupyterhub instance in NYU HSRN. If that is the case, that Dockerfile should not be in the root directory, it should be in a folder specifically for the demo. The Dockerfile in the root should contain only the necessary dependencies to run Alpha-AutoML. Also, I think PipelineProfiler doesn't work in jupyterlab.

Reply:
The current Dockerfile in the root folder is not specific to the HSRN cluster. It works on any installation of JupyterHub or in a single machine using docker (it starts the traditional Jupyter UI). This is why there are two 'flavors' of the image being built in the CI scripts (`ghcr.io/vida-nyu/alpha-automl` and `ghcr.io/vida-nyu/alpha-automl-jupyterhub`).

That said, I don't know why we are installing the dependencies like that with pinned versions. We actually should probably be just extending pre-built images with the latest versions of JupyterHub. In any case, I  think all of this should be discussed and changed in a separate PR. I would just revert it for now and we can discuss it the next meeting.

BTW, many people only use JupyterLab, so PipelineProfilter needs to be fixed too. It should be a high priority.

Reply:
I was not aware of `ghcr.io/vida-nyu/alpha-automl` and `ghcr.io/vida-nyu/alpha-automl-jupyterhub`. I agree, let's discuss this in another PR.